assayist/client/modify.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import itertools
+
+import neomodel
+
+from assayist.common.models.source import Component
+
+
+@neomodel.db.transaction
+def set_component_names(c_name, c_type, c_namespace='', alternatives=None):
+    """
+    Create, update, and/or merge components to ensure one canonical component remains.
+
+    :param str c_name: the canonical name of the component
+    :param str c_type: the canonical type of the component
+    :param str c_namespace: the canonical namespace of the component
+    :kwarg list alternatives: the alternative names associated with the component
+    :raises ValueError: if the input isn't the proper type or c_name's value is also in alternatives
+    """
+    for arg in (c_name, c_type, c_namespace):
```

Comment:
Should we also validate that name and type are not empty strings since the model accepts that as a valid value?

Reply:
I thought of that but `c_namespace` can be an empty string. I can add another for loop.